test-framework/examples/tests/src/test/java/org/keycloak/test/examples/CleanupTest.java

```diff
@@ -0,0 +1,75 @@
+package org.keycloak.test.examples;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.test.framework.annotations.InjectRealm;
+import org.keycloak.test.framework.annotations.KeycloakIntegrationTest;
+import org.keycloak.test.framework.realm.ManagedRealm;
+
+@KeycloakIntegrationTest
+public class CleanupTest {
+
+    @InjectRealm
+    ManagedRealm managedRealm;
+
+    @Test
+    public void dirty() {
+        checkCleanState();
+
+        managedRealm.dirty();
+
+        UserRepresentation userRepresentation = new UserRepresentation();
+        userRepresentation.setEnabled(true);
+        userRepresentation.setUsername("foobar");
+        managedRealm.admin().users().create(userRepresentation);
+    }
+
+    @Test
+    public void cleanupTask() {
+        checkCleanState();
+
+        managedRealm.cleanup().add(r -> r.users().list().forEach(u -> r.users().delete(u.getId()).close()));
```

Comment:
Is "r" a RealmResource?

Reply:
This is a lambda expression that implements `RealmUpdate`, so it's basically an implementation of `RealmUpdate#update(RealmConfigBuilder realm)`

Reply:
thank you for the explanation, I didn't know that Lambda can be used also like this, it that case users() method is missing in the RealmConfigBuilder class.